writer.go

```diff
@@ -55,30 +55,42 @@ type multiLevelWriter struct {
 }
 
 func (t multiLevelWriter) Write(p []byte) (n int, err error) {
+	var anyErr error
```

Comment:
It’s counter intuitive to use err declared in the prototype as a local variable and to create another one to be returned. Would be better to use _err to test individual errors and keep using err as the return value for instance.

Reply:
I've simplified the error variable usage, it was indeed messy, well noted.